rust-runtime/aws-smithy-http-server/src/routing/mod.rs

```diff
@@ -61,7 +65,7 @@ where
     /// Unless you add additional routes this will respond to `404 Not Found` to
     /// all requests.
     pub fn new() -> Self {
-        Self { routes: Default::default() }
+        Self { routes: Default::default(), request_specs: Default::default() }
```

Comment:
```suggestion
        Self { ..Default::default() }
```

Reply:
For reference: this change causes one test to abort badly with a stack overflow:

```
❯❯❯ cargo test                                                                                                      
    Blocking waiting for file lock on build directory
   Compiling aws-smithy-http-server v0.1.0 (/home/matbigoi/github/smithy-rs/rust-runtime/aws-smithy-http-server)
    Finished test [unoptimized + debuginfo] target(s) in 5.38s
     Running unittests (/home/matbigoi/github/smithy-rs/target/debug/deps/aws_smithy_http_server-fe9faed125bc0b03)

running 9 tests
test routing::into_make_service::tests::traits ... ok
test routing::route::tests::traits ... ok
test routing::request_spec::tests::repeated_query_keys_distinct_values_does_not_match ... ok
test routing::request_spec::tests::repeated_query_keys ... ok
test routing::request_spec::tests::repeated_query_keys_same_values_match ... ok
test routing::request_spec::tests::greedy_labels_match_greedily ... ok
test routing::request_spec::tests::empty_segments_at_the_end_do_matter ... ok
test routing::request_spec::tests::empty_segments_in_the_middle_dont_matter ... ok

thread 'routing::tests::simple_routing' has overflowed its stack
fatal runtime error: stack overflow
error: test failed, to rerun pass '--lib'

Caused by:
  process didn't exit successfully: `/home/matbigoi/github/smithy-rs/target/debug/deps/aws_smithy_http_server-fe9faed125bc0b03` (signal: 6, SIGABRT: process abort signal)
```

I am going to try to create a simple reproducer and file a bug report to Rustc

Reply:
RE: https://github.com/rust-lang/rust/issues/91012